src/main/java/uk/gov/hmcts/reform/slc/health/FtpHealthIndicator.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.slc.health;
+
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.services.FtpAvailabilityChecker;
+import uk.gov.hmcts.reform.slc.services.steps.sftpupload.FtpClient;
+
+import java.time.LocalTime;
+
+@Component
+public class FtpHealthIndicator extends AbstractHealthIndicator {
+
+    private final FtpAvailabilityChecker availabilityChecker;
+    private final FtpClient client;
+
+    public FtpHealthIndicator(FtpAvailabilityChecker availabilityChecker,
+                              FtpClient client) {
+        this.availabilityChecker = availabilityChecker;
+        this.client = client;
+    }
+
+    @Override
+    protected void doHealthCheck(Health.Builder builder) throws Exception {
+        if (availabilityChecker.isFtpAvailable(LocalTime.now())) {
+            client.isHealthy();
+        }
+
+        builder.up();
+    }
```

Comment:
There's a `HealthIndicator` interface you can implement instead of extending this class. `health()` method, that needs to be implemented, returns a value.

Reply:
For this indicator I do not want to use interface as I am not aware of any ambiguous exceptions client might spit out